src/grpc/stream_infer_handler.cc

```diff
@@ -596,7 +601,13 @@ ModelStreamInferHandler::StreamInferResponseComplete(
     void* userp)
 {
   State* state = reinterpret_cast<State*>(userp);
-
+  // Ignore Response from CORE in case GRPC Strict as we dont care about
+  if (state->context_->grpc_strict_) {
+    std::lock_guard<std::recursive_mutex> lock(state->context_->mu_);
+    if (state->context_->IsGRPCStrictError()) {
```

Comment:
Do you need both `grpc_stream_errror_state_` to be atomic and using `state->context_->mu_` everywhere you're accessing/modifying?

Reply:
I need grpc_stream_errror_state_ to be thread-safe so need the mutex.
Reverting grpc_stream_errror_state_ to bool

Reply:
should we limit the scope of the lock to only `if (state->context_->IsGRPCStrictError()) { ... } ` part?